pkg/flow/json_walker.go

```diff
@@ -0,0 +1,238 @@
+package flow
```

Comment:
The bulk of this logic is River-specific and can be moved to pkg/river so the other packages can remain internal.